src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditorCore.jsx

```diff
@@ -0,0 +1,239 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { get } from 'lodash-es';
+
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import NewChildIcon from '@material-ui/icons/AddCircle';
+import TextInput from '../../../../components/inputs/TextInput';
+import DeleteButton from '../../../../components/DeleteButton';
+import Button from '../../../../components/Button';
+import Text from '../../../../components/Text';
+
+function deleteCategory(allRelationships, category) {
+  delete allRelationships[category];
+  return allRelationships;
+}
+
+function deleteRelationship(
+  allRelationships,
+  currentCategory,
+  relationship,
+) {
+  const targetRelationshipCategory = get(
+    allRelationships,
+    currentCategory,
+    [],
+  );
+  const relationshipsWithoutTarget = targetRelationshipCategory.filter(
```

Comment:
This assumes no duplicate relationships, which is not a safe assumption at this point since deduplication does not happen until later. If I have relationships: ['relationship 1', 'relationship 2', 'relationship 1'] and I delete a row with 'relationship 1' in it, both entries will be deleted.